compiler/rustc_mir/src/interpret/intern.rs

```diff
@@ -105,7 +106,7 @@ fn intern_shallow<'rt, 'mir, 'tcx, M: CompileTimeMachine<'mir, 'tcx>>(
     // changes in this function.
     match kind {
         MemoryKind::Stack
-        | MemoryKind::ConstHeap
+        | MemoryKind::Machine(_)
```

Comment:
Please explicitly use `const_eval::MemoryKind::Heap` here instead of `_`, so any future changes will affect this code again

Reply:
force pushed